### PR TITLE
[RF] Refactoring of all tests <Snegafal>

### PR DESCRIPTION
**Pull request by Snegafal**
Refactored all tests 
RF - https://trello.com/c/l87OHZUN/905-rf-refactoring-all-tests-snegafal

### CODE REVIEW

**Review by WeraStremedlowska — APPROVED** [review]

File: src/test/java/SnegafalTest.java [hunk 1]
```diff
@@ -1,137 +1,96 @@
...
+    private final By START_PAGE_TEMPERATURE = By.xpath("//span[@class='heading']");
+    private final By DEGREE_SIGN_IN_DIFFERENT_WEATHER_POPUP = By.xpath("//div[@class='input-with-selection']/div");
+    private final By DATA_SOURCE_SELECTOR = By.xpath("//div[@class='dropdown-selector']");
+    private final By DATA_SOURCE_OPTIONS = By.xpath("//div[@class='owm-selector open']//ul[@class='dropdown-menu']/li");
+    private final By ACTIVE_ICON_IN_DIFFERENT_WEATHER_POP_UP = By.className("activeIcon");
+
+    @Test
+    public void testOnlyOneIconIsHighlightedWhenDifferentWeatherPopupAppears() {
+        openBaseURL();
+
```
Comment (larisabratukhina) [inline src/test/java/SnegafalTest.java:33]: delete line

File: src/test/java/SnegafalTest.java [hunk 2]
```diff
@@ -1,137 +1,96 @@
...
+        click(DIFFERENT_WEATHER_BUTTON);
+        waitElementToBeVisible(DIFFERENT_WEATHER_POP_UP);
+        int activeIcon = seeAllElementAndCount(ACTIVE_ICON_IN_DIFFERENT_WEATHER_POP_UP);
```
Comment (larisabratukhina) [inline src/test/java/SnegafalTest.java:36]: add line

File: src/test/java/SnegafalTest.java [hunk 3]
```diff
@@ -1,137 +1,96 @@
...
+    @Test
+    public void testVerifyTemperatureInDifferentPopUpWeatherAndInStartPage() {
+        openBaseURL();
+
+        String temperatureInStartPage = getText(START_PAGE_TEMPERATURE);
+
+        click(DIFFERENT_WEATHER_BUTTON);
+        waitElementToBeVisible(DIFFERENT_WEATHER_POP_UP);
+        click(MORE_OPTIONS_BUTTON);
+        String temperatureInPopup = getTextByAttribute(DIFFERENT_WEATHER_TEMPERATURE_INPUT, "_value");
```
Comment (larisabratukhina) [inline src/test/java/SnegafalTest.java:53]: add line

File: src/test/java/SnegafalTest.java [hunk 4]
```diff
@@ -1,137 +1,96 @@
...
+        String degreeSign = getText(DEGREE_SIGN_IN_DIFFERENT_WEATHER_POPUP);
+        String temperatureInPopupWithDegree = temperatureInPopup.concat(degreeSign);
+
+        Assert.assertEquals(temperatureInPopupWithDegree, temperatureInStartPage);
+    }
+
+    @Test
+    public void testIncreaseTemperatureInDifferentWeatherPopUpByPressingAKey() {
+        openBaseURL();
+
```
Comment (larisabratukhina) [inline src/test/java/SnegafalTest.java:63]: delete line

File: src/test/java/SnegafalTest.java [hunk 5]
```diff
@@ -1,137 +1,96 @@
...
+        int increasedTemperatureInInput = Integer.parseInt(
+                getTextByAttribute(DIFFERENT_WEATHER_TEMPERATURE_INPUT, "_value"));
+
+        Assert.assertEquals(increasedTemperatureInInput - temperatureInPopup, 1);
+    }
+
+    @Test
+    public void testNamesInDataSourceSelectorInDifferentWeatherPopUp() {
+
+        openBaseURL();
```
Comment (larisabratukhina) [inline src/test/java/SnegafalTest.java:80]: delete line

File: src/test/java/SnegafalTest.java [hunk 6]
```diff
@@ -1,137 +1,96 @@
...
+
+    @Test
+    public void testIncreaseTemperatureInDifferentWeatherPopUpByPressingAKey() {
+        openBaseURL();
+
+        click(DIFFERENT_WEATHER_BUTTON);
+        waitElementToBeVisible(DIFFERENT_WEATHER_POP_UP);
+        click(MORE_OPTIONS_BUTTON);
+        WebElement temperatureInput = getDriver().findElement(DIFFERENT_WEATHER_TEMPERATURE_INPUT);
+        int temperatureInPopup = Integer.parseInt(getTextByAttribute(
```
Comment (larisabratukhina) [inline src/test/java/SnegafalTest.java:68]: add line

File: src/test/java/SnegafalTest.java [hunk 7]
```diff
@@ -1,137 +1,96 @@
...
+                DIFFERENT_WEATHER_TEMPERATURE_INPUT, "_value"));
+        temperatureInput.sendKeys(Keys.ARROW_UP);
+        int increasedTemperatureInInput = Integer.parseInt(
```
Comment (larisabratukhina) [inline src/test/java/SnegafalTest.java:71]: add line

File: src/test/java/SnegafalTest.java [hunk 8]
```diff
@@ -1,137 +1,96 @@
...
+    public void testNamesInDataSourceSelectorInDifferentWeatherPopUp() {
+
+        openBaseURL();
+
+        click(DIFFERENT_WEATHER_BUTTON);
+        waitElementToBeVisible(DIFFERENT_WEATHER_POP_UP);
+        click(MORE_OPTIONS_BUTTON);
+        click(DATA_SOURCE_SELECTOR);
+
+        List<WebElement> dataSourceElements = getDriver().findElements(DATA_SOURCE_OPTIONS);
```
Comment (elviraKh1) [inline src/test/java/SnegafalTest.java:87]: 87 line could be change with    List<WebElement> dataSourceElements = getListOfElements(DATA_SOURCE_OPTIONS);

File: src/test/java/SnegafalTest.java [hunk 9]
```diff
@@ -1,137 +1,96 @@
...
+        List <String> actualDataSourceElements = new ArrayList<>();
+        for (WebElement option : dataSourceElements) {
+            actualDataSourceElements.add(option.getText());
+        }
```
Comment (elviraKh1) [inline src/test/java/SnegafalTest.java:91]: 88-91 lines could be moved to BaseTest as general method and  returns List<String> of WebElement texts

**Pull request merged**